src/main/java/frc/robot/classes/TimeOfFlightSensor.java

```diff
@@ -0,0 +1,239 @@
+    int value;
+  }
+
+  int parseIntFromIndex(SingleCharSequence charSeq, int readLen, IntRef lastComma) {
+    int nextComma = 0;
+    try {
+      nextComma = findNextComma(charSeq.data, readLen, lastComma.value);
+      int value = Integer.parseInt(charSeq, lastComma.value + 1, nextComma, 10);
+      lastComma.value = nextComma;
+      return value;
+    } catch (Exception ex) {
+      return 0;
+    }
+  }
+
+  private int findNextComma(byte[] data, int readLen, int lastComma) {
+    while (true) {
+      if (readLen <= lastComma + 1 ) {
+        return readLen;
+      }
+      lastComma++;
+      if (data[lastComma] == ',') {
+        break;
+      }
+    }
+    return lastComma;
+  }
+
+  private final AtomicBoolean debugPrints = new AtomicBoolean();
+  private boolean hasDistance0;
+  private boolean hasDistance1;
+  private final RawDistance distance0 = new RawDistance();
+  private final RawDistance distance1 = new RawDistance();
+  private double lastReadTime;
+  private final ReentrantLock threadLock = new ReentrantLock();
+  private final Thread readThread;
+  private final AtomicBoolean threadRunning = new AtomicBoolean(true);
+
+
+  private void threadMain() {
+    // Using JNI for a non allocating read
+    int port = SerialPortJNI.serialInitializePort((byte)1);
+    SerialPortJNI.serialSetBaudRate(port, 115200);
+    SerialPortJNI.serialSetDataBits(port, (byte)8);
+    SerialPortJNI.serialSetParity(port, (byte)0);
+    SerialPortJNI.serialSetStopBits(port, (byte)10);
+
+    SerialPortJNI.serialSetTimeout(port, 1);
+    SerialPortJNI.serialEnableTermination(port, '\n');
+    
+    //int port = SerialPortJNI.serialInitializePort((byte)3);
+    HAL.report(tResourceType.kResourceType_SerialPort, 2);
+
+    byte[] buffer = new byte[257];
+    SingleCharSequence charSeq = new SingleCharSequence();
+    charSeq.data = buffer;
+    IntRef lastComma = new IntRef();
+
+    RawDistance distance0 = new RawDistance();
+    RawDistance distance1 = new RawDistance();
+
+    while (threadRunning.get()) {
+      int read = SerialPortJNI.serialRead(port, buffer, buffer.length - 1);
+      if (read <= 0) {
+        try {
+          threadLock.lock();
+          this.hasDistance0 = false;
+          this.hasDistance1 = false;
+        } finally {
+          threadLock.unlock();
+        }
+        continue;
+      }
+      if (!threadRunning.get()) {
+        break;
+      }
+
+      // Trim trailing newline if exists
+      if (buffer[read - 1] == '\n') {
+        read--;
+      }
+
+      if (read == 0) {
+        continue;
+      }
+
+      lastComma.value = -1;
+
+      //boolean hasDistance0 = parseIntFromIndex(charSeq, read, lastComma) != 0;
+      //boolean hasDistance1 = parseIntFromIndex(charSeq, read, lastComma) != 0;
+      distance0.classDistance = parseIntFromIndex(charSeq, read, lastComma);
+      distance1.classDistance = parseIntFromIndex(charSeq, read, lastComma);
+      boolean hasDistance0 = distance0.classDistance != -1;
```

Comment:
Let's make this <=0

I don't know why the second sensor is showing 0 right now.  But that should trigger a failure as well, because there is no way to get the cone close enough to the sensor to register 0